### ISSUE TITLE
Windows build of master currently failing

### ISSUE THREAD

**Issue by sothix** [borgbackup/borg#8381]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

BUG

#### Your borg version (borg -V).

Master

## Describe the problem you're observing.

During the build it fails to compile:
```
Collecting cryptography<43.0.0 (from borgstore~=0.0.1->borgbackup==2.0.0b11.dev3+gc278a8e8)
Using cached cryptography-42.0.8.tar.gz (671 kB)
Installing build dependencies ... error
...
        gcc -DNDEBUG -g -fwrapv -O3 -Wall -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -O3 -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-st
rong -O3 -DFFI_BUILDING=1 -IC:/msys64/ucrt64/include/python3.11 -c src/c/_cffi_backend.c -o build/temp.mingw_x86_64_ucrt-cpython-311/src/c/_cffi_backend.o
        src/c/_cffi_backend.c: In function 'b_do_dlopen':
        src/c/_cffi_backend.c:4568:40: error: passing argument 1 of 'PyUnicode_AsWideChar' from incompatible pointer type [-Wincompatible-pointer-types]
         4568 |             sz1 = PyUnicode_AsWideChar((PyUnicodeObject *)filename_unicode,
              |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              |                                        |
              |                                        PyUnicodeObject *
        In file included from C:/msys64/ucrt64/include/python3.11/Python.h:51,
                         from src/c/_cffi_backend.c:2:
        C:/msys64/ucrt64/include/python3.11/unicodeobject.h:299:15: note: expected 'PyObject *' {aka 'struct _object *'} but argument is of type 'PyUnicodeObject *'
          299 |     PyObject *unicode,          /* Unicode object */
              |     ~~~~~~~~~~^~~~~~~
        error: command 'C:\\msys64\\ucrt64\\bin/gcc.EXE' failed with exit code 1
        [end of output]
```

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes 
new install of MSYS2.
```
pacman -S git
git clone https://github.com/borgbackup/borg.git 
cd borg
./scripts/msys2-install-deps 
pip install -e .
```
It can be fixed by:
`pacman -S mingw-w64-ucrt-x86_64-python-paramiko`

or including `mingw-w64-ucrt-x86_64-python-paramiko` in the `./scripts/msys2-install-deps` 

Also export `SETUPTOOLS_USE_DISTUTILS=stdlib` didn't seem to be required.

**Comment by ThomasWaldmann** [comment]
Might be.

Windows is not being CI tested anymore, because of #8264 still searching for someone who fixes it.

**Comment by ThomasWaldmann** [comment]
@sothix If you can debug such stuff, try fixing the windows CI, PRs welcome!

**Comment by ThomasWaldmann** [comment]
The fundamental issues were fixed by #8382.

> Also export SETUPTOOLS_USE_DISTUTILS=stdlib didn't seem to be required.

That is still TODO / to review.
